pandas/core/index.py

```diff
@@ -1179,17 +1179,18 @@ def argsort(self, *args, **kwargs):
         return result.argsort(*args, **kwargs)
 
     def __add__(self, other):
-        if isinstance(other, Index):
+        if com.is_list_like(other):
             warnings.warn("using '+' to provide set union with Indexes is deprecated, "
                           "use '|' or .union()",FutureWarning)
+        if isinstance(other, Index):
```

Comment:
@jreback I don't really see how this is fixed? It is still

````
if com.is_list_like(other):
    warnings.warn(...)
if isinstance(other, Index):
    return self.union(other)
```

So a list will trigger the warning, but not do the union set operation.